crates/forge/tests/data/contracts/spy_events_checker.cairo

```diff
@@ -74,5 +76,9 @@ mod SpyEventsChecker {
             self.emit(Event::SecondEvent(SecondEvent { some_data, some_more_data }));
             self.emit(Event::ThirdEvent(ThirdEvent { some_data, some_more_data, even_more_data }));
         }
+
+        fn emit_event_syscall(ref self: ContractState, some_data: felt252) {
+            starknet::emit_event_syscall(array![].span(), array![some_data].span()).unwrap_syscall();
+        }
```

Comment:
We could emit some keys to check if we handle them correctly too

Reply:
The first key will be interpreted as a hashed name. There is no other option to handle named events

Reply:
Why is there a field in `UnnamedEvent` struct for keys then? That seems counterintuitive. Perhaps we should serialize the events in the same way in cairo (with a uniform `EventData`) and instead serialize the passed one, to the uniform data. We could then serialize the user-passed event into uniform representation, and then compare with ones that were captured. WDYT?